packages/compass-aggregations/src/utils/stage.js

```diff
@@ -42,92 +40,98 @@ export const parseNamespace = (currentDb, stage) => {
+const isSupportedVersion = ({ operatorVersion, serverVersion }) =>
+  semver.gte(serverVersion, operatorVersion);
 
 /**
  * Is search on a view, time-series, or regular collection?
  *
  * @param {Object} options - The options to exclude the full-text search stages on views.
- * @property {String} oName - The stage name.
+ * @property {String} operatorName - The stage name.
  * @property {Boolean} isTimeSeries - The isTimeSeries flag.
  * @property {Boolean} isReadonly - The isReadonly flag.
  * @property {String} sourceName - The namespace on which created the view.
  *
  * @returns {boolean} If search on a view, time-series, or regular collection.
  */
-const isSearchOnView = ({ oName, isTimeSeries, isReadonly, sourceName }) =>
-  [SEARCH, SEARCH_META, DOCUMENTS].includes(oName) &&
+const isSearchOnView = ({ operatorName, isTimeSeries, isReadonly, sourceName }) =>
+  [SEARCH, SEARCH_META, DOCUMENTS].includes(operatorName) &&
   (isTimeSeries || (isReadonly && !!sourceName));
 
 /**
  * Filters stage operators by server version.
  *
  * @param {Object} options - Info about the server and the collection to filter agg stages.
  * @property {String} version - The current server version.
- * @property {boolean} allowWrites - If writes are allowed.
  * @property {String} env - The current env.
  * @property {boolean} isTimeSeries - The isTimeSeries flag.
  * @property {boolean} isReadonly - The isReadonly flag.
  * @property {String} sourceName - The namespace on which created the view.
  *
  * @returns {Array} Stage operators supported by the current version of the server.
  */
-export const filterStageOperators = ({ serverVersion, allowWrites, env, isTimeSeries, isReadonly, sourceName }) => {
-  const parsedVersion = semver.parse(serverVersion);
-  const cleanVersion = parsedVersion
-    ? [parsedVersion.major, parsedVersion.minor, parsedVersion.patch].join('.')
+export const filterStageOperators = ({ serverVersion, env, isTimeSeries, isReadonly, sourceName }) => {
+  const serverSemver = semver.parse(serverVersion);
+  const cleanServerVersion = serverSemver
+    ? [serverSemver.major, serverSemver.minor, serverSemver.patch].join('.')
     : serverVersion;
 
-  return STAGE_OPERATORS.filter((o) => {
-    if (isNotWritable({ oName: o.name, allowWrites })) return false;
-    if (isSearchOnView({ oName: o.name, isTimeSeries, isReadonly, sourceName })) return false;
-    if (o.dbOnly) return false;
+  return STAGE_OPERATORS.filter((operator) => {
+    if (isSearchOnView({
+      operatorName: operator.name,
+      isTimeSeries,
+      isReadonly,
+      sourceName
+    })) return false;
+    if (operator.dbOnly) return false;
 
-    return isSupportedVersion({ oVersion: o.version, version: cleanVersion }) &&
-      isSupportedEnv({ oEnv: o.env, env }) ||
-      isAtlasOnly(o.env);
+    return isSupportedVersion({
+      operatorVersion: operator.version,
+      serverVersion: cleanServerVersion
+    }) &&
+      isSupportedEnv({
```

Comment:
This is a pass-by change, I moved `isAtlasOnly` inside `isSupportedEnv` so even `isAtlasOnly` stages will be filtered by version. 

We often use the server version to hide stages that will be released in the future, I thought we don't want to ignore the version here. Happy to revert if this was intentional.